internal/rbd/controllerserver.go

```diff
@@ -794,6 +795,50 @@ func (cs *ControllerServer) checkErrAndUndoReserve(
 	return &csi.DeleteVolumeResponse{}, nil
 }
 
+// isMigrationVolHandle check the passed in volHash is a volume handle
+// of migrated volume.
+func isMigrationVolHandle(volHash string) bool {
+	if strings.Contains(volHash, migVersion) &&
+		strings.Contains(volHash, migImageNamePfx) {
+		return true
+	}
+
+	return false
+}
+
+// parseMigrationVolHandle decode the volume handle and generate a migrationHandle
+// struct which consists of version, mon, image name and pool information.
+func parseMigrationVolHandle(vh string) (*migrationHandle, error) {
+	mh := new(migrationHandle)
+	handSlice := strings.Split(vh, migHandleFieldSep)
+	if len(handSlice) < migHandleLength {
+		// its short of length in this case, so return error
+		return nil, ErrInvalidVolID
+	}
+
+	// Store pool
+	mh.Pool = strings.Join(handSlice[3:], migHandleFieldSep)
+
+	// Parse migration version, mons and image
+	for _, id := range handSlice[:3] {
+		fmt.Println(id)
+		switch {
+		case id == migVersion:
+			mh.Version = id
+		case strings.Contains(id, migImageNamePfx):
+			mh.ImageName = id
+		case strings.Contains(id, migMonPfx):
+			newMons := strings.Trim(id, migMonPfx)
+			mh.Mons = newMons
+		}
+	}
+	if mh.ImageName == "" || mh.Pool == "" || mh.Mons == "" {
```

Comment:
Can we return separate warnings so that we can identify what is empty in logs?
E.g: if mons information is not present, passing "monitor information can not be empty" will give a more clear idea than a general ErrInvalidVolID

Reply:
Done. thanks.